pkg/services/services.go

```diff
@@ -34,15 +44,79 @@ func New(httpClient *resty.Client, logger *slog.Logger, config configuration.Wal
 		panic("httpClient is required")
 	}
 
+	woc := whatsonchain.New(httpClient, logger, config.Chain, config.WhatsOnChain)
+
+	rawTxResultServices := servicequeue.New[RawTxResultService]().Add(servicequeue.ServiceExecutor[RawTxResultService]{
+		Name:    "WhatsOnChain",
+		Service: woc,
+	})
+
 	return &WalletServices{
-		httpClient:   httpClient,
-		chain:        config.Chain,
-		config:       &config,
-		logger:       logger,
-		whatsonchain: whatsonchain.New(httpClient, logger, config.Chain, config.WhatsOnChain),
+		httpClient:    httpClient,
+		chain:         config.Chain,
+		config:        &config,
+		logger:        logger,
+		whatsonchain:  woc,
+		rawTxServices: rawTxResultServices,
 	}
 }
 
+// RawTx attempts to obtain the raw transaction bytes associated with a 32 byte transaction hash (txid).
+//
+// Cycles through configured transaction processing services attempting to get a valid response.
+//
+// On success:
+// Result txid is the requested transaction hash
+// Result rawTx will be an array containing raw transaction bytes.
+// Result name will be the responding service's identifying name.
+// Returns result without incrementing active service.
+//
+// On failure:
+// Result txid is the requested transaction hash
+// Result mapi will be the first mapi response obtained (service name and response), or null
+// Result error will be the first error thrown (service name and CwiError), or null
+// Increments to next configured service and tries again until all services have been tried.
+func (s *WalletServices) RawTx(txID string) (internal.RawTxResult, error) {
+	var lastErr error
+
+	for tries := 0; tries < s.rawTxServices.Count(); tries++ {
+		srv, err := s.rawTxServices.Current()
+		if err != nil {
+			return internal.RawTxResult{}, fmt.Errorf("error while getting current service: %w", err)
+		}
+
+		resp, err := srv.Service.RawTx(txID, s.chain)
+		if err != nil {
+			lastErr = fmt.Errorf("error while getting raw tx result from service: %s => %w", srv.Name, err)
+			s.logger.Error(lastErr.Error())
+			s.rawTxServices.Next()
+			continue
+		}
+
+		if resp == nil {
+			lastErr = fmt.Errorf("transaction with txID: %s not found", txID)
+			s.rawTxServices.Next()
+			continue
+		}
+
+		hash := utils.DoubleSha256BE(resp.RawTx)
+		encodedHex := hex.EncodeToString(hash)
```

Comment:
maybe instead of some DoubleSha256BE function, create a function `txutil.TransactionIDFromRawTx()`